### PR TITLE
adapt integer comparison tests for LLVM 20 IR changes

### PR DESCRIPTION
**Pull request by krasimirgg**
The LLVM commit https://github.com/llvm/llvm-project/commit/abf69a167bbc99054871e3f9cc8810bbebcb6747 changed the IR in a few comparison tests:
https://buildkite.com/llvm-project/rust-llvm-integrate-prototype/builds/30500#01917017-26fe-4a4d-956b-725a2903e5a8

Adapted accordingly.

@rustbot label: +llvm-main
r? @nikic

### CODE REVIEW

File: tests/codegen/comparison-operators-2-tuple.rs [hunk 1]
```diff
@@ -1,5 +1,8 @@
 //@ compile-flags: -C opt-level=1 -Z merge-functions=disabled
 //@ only-x86_64
+//@ revisions: llvm-pre-20 llvm-20
+//@[llvm-20]min-llvm-version: 20
```
Comment (nikic) [inline tests/codegen/comparison-operators-2-tuple.rs:4]:
```suggestion
//@ [llvm-20] min-llvm-version: 20
```
etc

File: tests/codegen/comparison-operators-2-tuple.rs [hunk 2]
```diff
@@ -77,11 +80,17 @@ pub fn check_ge_direct(a: TwoTuple, b: TwoTuple) -> bool {
...
-    // CHECK-DAG: %[[EQ:.+]] = icmp eq i16 %[[A0]], %[[B0]]
-    // CHECK-DAG: %[[CMP0:.+]] = icmp slt i16 %[[A0]], %[[B0]]
-    // CHECK-DAG: %[[CMP1:.+]] = icmp ult i16 %[[A1]], %[[B1]]
-    // CHECK: %[[R:.+]] = select i1 %[[EQ]], i1 %[[CMP1]], i1 %[[CMP0]]
-    // CHECK: ret i1 %[[R]]
+    // llvm-20-DAG: %[[CMP0:.+]] = tail call noundef range(i8 -1, 2) i8 @llvm.scmp.i8.i16(i16 %[[A0]], i16 %[[B0]])
+    // llvm-20-DAG: %[[EQ:.+]] = icmp eq i16 %[[A0]], %[[B0]]
+    // llvm-20-DAG: %[[CMP1:.+]] = tail call range(i8 -1, 2) i8 @llvm.ucmp.i8.i16(i16 %[[A1]], i16 %[[B1]])
+    // llvm-20: %[[S:.+]] = select i1 %[[EQ]], i8 %[[CMP1]], i8 %[[CMP0]]
+    // llvm-20: %[[R:.+]] = icmp eq i8 %[[S]], -1
```
Comment (nikic) [inline tests/codegen/comparison-operators-2-tuple.rs:87]: This is a regression. I'll take a look.

**Comment by krasimirgg** [comment]
@nikic also could you suggest how to best handle the `tidy: line longer than 100 chars` errors? Possibly disabling it for specific lines? I could look into adapting the CHECKs to spread them on multiple lines, but that seems like it will reduce readability. Also it seems there are already pre-existing CHECK lines over 100 chars, e.g.,: `// CHECK-SAME: (i16 noundef %[[A0:.+]], i16 noundef %[[A1:.+]], i16 noundef %[[B0:.+]], i16 noundef %[[B1:.+]])`.

**Comment by nikic** [comment]
@krasimirgg You can add `//@ ignore-tidy-linelength`, it's pretty common for codegen tests.

**Comment by krasimirgg** [comment]
@nikic FYI, looks like this LLVM commit also changes the assembly output of `tests/assembly/x86_64-cmp.rs#OPTIM`: https://buildkite.com/llvm-project/rust-llvm-integrate-prototype/builds/30520#01917461-b781-4d17-80e7-85cd0a8681d8

File: tests/codegen/comparison-operators-2-tuple.rs (hunk 2 above)
Comment (nikic) [reply tests/codegen/comparison-operators-2-tuple.rs:87]: https://github.com/llvm/llvm-project/pull/105536 should fix this.

**Comment by nikic** [comment]
> @nikic FYI, looks like this LLVM commit also changes the assembly output of `tests/assembly/x86_64-cmp.rs#OPTIM`: https://buildkite.com/llvm-project/rust-llvm-integrate-prototype/builds/30520#01917461-b781-4d17-80e7-85cd0a8681d8

New asm looks good, so should update that test with revision as well.

**Comment by krasimirgg** [comment]
> @krasimirgg You can add `//@ ignore-tidy-linelength`, it's pretty common for codegen tests.

Thanks! Had to do it without the `@`, otherwise I saw an `error: detected unknown compiletest test directive `ignore-tidy-linelength` in /tmp/rustllvmhead/build/rust/tests/codegen/comparison-operators-2-tuple.rs:1`

File: tests/codegen/comparison-operators-2-tuple.rs (hunk 2 above)
Comment (krasimirgg) [reply tests/codegen/comparison-operators-2-tuple.rs:88]: Nice! I'll wait for this to get submitted and update this PR.

Comment (nikic) [reply tests/codegen/comparison-operators-2-tuple.rs:88]: This is merged now.

Comment (krasimirgg) [reply tests/codegen/comparison-operators-2-tuple.rs]: Nice! This eliminated the changes needed to one of the tests.

**Pull request merged**